shared/services/rewards/utils.go

```diff
@@ -305,12 +304,23 @@ func DownloadRewardsFile(cfg *config.RocketPoolConfig, interval uint64, cid stri
 				}
 			}
 
+			// Verify if the downloaded file has the expected CID
+			calculatedCid, err := SingleFileDirIPFSCid(writeBytes, rewardsTreePath)
```

Comment:
`writeBytes` here is always decompressed, but the ipfs cid will be on the .zst-suffixed, compressed file.

We should deserialize the decompressed bytes into a RewardsFIle / MinipoolPerformanceFile struct and check the merkle root instead- standardizing on a specific zstd implementation (that specifically says _not to hash its output for similarity_) is going to be fragile.

Reply:
(
ideally we recompute the root so we are certain that:
1. every proof is correct
2. every proof is present
)